scalyr_agent/log_processing.py

```diff
@@ -1447,6 +1453,36 @@ def __init__(self, file_path, config, log_config, close_when_staleness_exceeds=N
 
         self.__disable_processing_new_bytes = config.disable_processing_new_bytes
 
+    def finish( self ):
```

Comment:
Change this to `close_when_caught_up`.  I hate that name, but conveys the idea.  Other alternatives are `close(when_caught_up=False)` (so the idea of being caught is just optional.. but that would imply we support the case `when_caught_up=False` which would require more coding.